java/code/src/com/suse/manager/ssl/SSLCertData.java

```diff
@@ -139,12 +140,10 @@ public int hashCode() {
      * @return machine name
      */
     public String getMachineName() {
-        List<String> dataList = Arrays.asList(this.getCn().split("\\."));
-        if (dataList.size() > 2) {
-            return String.join(".", dataList.subList(0, dataList.size() - 2));
-        }
-        else {
-            return String.join(".");
-        }
+        String[] hostnameParts = this.getCn().replace("*", "_star_").split("\\.");
```

Comment:
What I don't understand here is why this is needed... Was the bug only if there is a `*` in the proxy FDQN? or is that character added somehow? The proxy FQDN should not contain such a character: it's a single name.

Reply:
I am not sure why you'd use asterisk, ie, specifying/restricting domain in this context.
I wasn't expecting this to be a valid input, it doesn't even fail any validation. For me, this could have been a slip.
But the fact they were specifically handled by python's machine name mapping is what made me think there should be a reason for letting it be. 

As far as i see we have 2 options:
- keep the solution, focusing on the original of accessing the uploaded files
- add a validation to the fqdn inputs 'Proxy FQDN' and 'Parent FQDN' 

what do you think?

Reply:
Adding validation of the FQDNs couldn't be a bad thing, but I had the issue here even without any invalid one... so I think the truth is somewhere else